ros2_batch_job/__main__.py

```diff
@@ -521,15 +521,20 @@ def run(args, build_function, blacklisted_package_names=None):
     with change_directory(args.workspace):
         print('# BEGIN SUBSECTION: install Python packages')
         # Update setuptools
-        job.run(['"%s"' % job.python, '-m', 'pip', 'install', '-U', 'pip', 'setuptools'],
-                shell=True)
+        pip_cmd = ['"%s"' % job.python, '-m', 'pip', 'install', '-U']
+        if args.do_venv or sys.platform == 'win32':
+            # Force reinstall so all dependencies are in virtual environment
+            # On Windows since we switch between the debug and non-debug
+            # interpreter all packages need to be reinstalled too
+            pip_cmd.append('--force-reinstall')
+        job.run(pip_cmd + ['pip', 'setuptools',], shell=True)
         # Print setuptools version
         job.run(['"%s"' % job.python, '-c', '"import setuptools; print(setuptools.__version__)"'],
                 shell=True)
         # Print the pip version
         job.run(['"%s"' % job.python, '-m', 'pip', '--version'], shell=True)
         # Install pip dependencies
-        pip_packages = list(pip_dependencies)
+        pip_packages = ['pip', 'setuptools', ] + list(pip_dependencies)
```

Comment:
I believe we were intentionally updating pip and setuptools before updating other packages so that the remaining packages were installed using the latest version of these utilities.

Reply:
That's a good point, and they still are in line 530. This workaround came from the issue that --force-reinstall on the pip_dependencies list caused setuptools to be reinstalled as well. But for some reason, the pip wheel was not found if they were done separately. This was the only work around I could find.

Reply:
It looks like the error I was running into no longer occurs.

Reply:
Reverting this section